### PR TITLE
Support for additional S3 storage classes

### PR DESCRIPTION
**Pull request by mdenomy**
Full list of supported storage classes 

* standard
* intelligent_tiering
* standard_ia
* onezone_ia
* glacier
* glacier_ir
* deep_archive
* reduced_redundancy

### CODE REVIEW

File: pkg/commands/logging/s3/create.go [hunk 1]
```diff
@@ -216,6 +214,30 @@ func (c *CreateCommand) ConstructInput(serviceID string, serviceVersion int) (*f
 	return &input, nil
 }
 
+func ValidateRedundancy(val string) (redundancy fastly.S3Redundancy, err error) {
```
Comment (mdenomy) [inline pkg/commands/logging/s3/create.go:217]: Though it's not a lot of duplication, I didn't want to repeat this for the update command. If there is a more idiomatic go way of extracting this logic, I'd be happy to make any changes.

File: pkg/commands/logging/s3/create.go [hunk 2]
```diff
@@ -216,6 +214,30 @@ func (c *CreateCommand) ConstructInput(serviceID string, serviceVersion int) (*f
...
+	case string(fastly.S3RedundancyGlacierInstantRetrieval):
+		redundancy = fastly.S3RedundancyGlacierInstantRetrieval
+	case string(fastly.S3RedundancyGlacierFlexibleRetrieval):
+		redundancy = fastly.S3RedundancyGlacierFlexibleRetrieval
+	case string(fastly.S3RedundancyGlacierDeepArchive):
+		redundancy = fastly.S3RedundancyGlacierDeepArchive
+	case string(fastly.S3RedundancyReduced):
+		redundancy = fastly.S3RedundancyReduced
+	default:
+		err = fmt.Errorf("Unknown redundancy: " + val)
```
Comment (Integralist) [inline pkg/commands/logging/s3/create.go:236]: Error messages in go should be lowercase. 

```suggestion
		err = fmt.Errorf("unknown redundancy: " + val)
```

File: pkg/commands/logging/s3/create.go (hunk 1 above)
Comment (Integralist) [inline pkg/commands/logging/s3/create.go:217]: As noted by the linter, can we add a comment...
```suggestion
// ValidateRedundancy identifies the given redundancy type.
func ValidateRedundancy(val string) (redundancy fastly.S3Redundancy, err error) {
```

**Review by Integralist — CHANGES_REQUESTED** [review]

File: pkg/commands/logging/s3/create.go (hunk 1 above)
Comment (mdenomy) [reply pkg/commands/logging/s3/create.go:218]: Thanks, added it in https://github.com/fastly/cli/pull/641/commits/7ad861493108308ff40c5d3cc4b5f6ac59c0c536

File: pkg/commands/logging/s3/create.go (hunk 2 above)
Comment (mdenomy) [reply pkg/commands/logging/s3/create.go]: Added in https://github.com/fastly/cli/pull/641/commits/7ad861493108308ff40c5d3cc4b5f6ac59c0c536

**Review by Integralist — APPROVED** [review]

**Comment by Integralist** [comment]
Thanks @mdenomy this looks great 👍🏻

**Pull request merged**